pkg/kudo/instance.go

```diff
@@ -61,13 +64,44 @@ func ListInstances(client client.Client, namespace string) ([]Instance, error) {
 	return instances, nil
 }
 
+func currentPlanUID(instance Instance, plan string) apimachinerytypes.UID {
+	if _, ok := instance.Status.PlanStatus[plan]; !ok {
+		// The plan may not have been in use before.
+		// We continue, assuming that the plan name is valid and present in OperatorVersion.
+		return ""
+	}
+
+	ps := instance.Status.PlanStatus[plan]
+
+	return ps.UID
+}
+
 func currentPlanStatusAndMessage(instance Instance, plan string) (kudov1beta1.ExecutionStatus, string) {
 	if _, ok := instance.Status.PlanStatus[plan]; !ok {
 		// The plan may not have been in use before.
 		// We continue, assuming that the plan name is valid and present in OperatorVersion.
 		return kudov1beta1.ExecutionNeverRun, ""
 	}
 
+	ps := instance.Status.PlanStatus[plan]
+	if ps.Status != kudov1beta1.ExecutionFatalError || ps.Message != "" {
+		return ps.Status, ps.Message
+	}
+
+	for _, phaseStatus := range ps.Phases {
```

Comment:
Please add a comment why this is necessary. I'd expect that `PlanStatus[plan].Status` is always the source of truth. This indicates that there are conditions where that isn't the case. Is this a bug in KUDO?

Reply:
Not sure if it's a bug or a feature, but in a lot of cases the full message is only available in the deepest status item:
https://github.com/kudobuilder/kudo/blob/b1ef318016ad85eaa46bcf00ad23d234a524d609/pkg/engine/workflow/engine.go#L155

Reply:
It kind of makes sense if you just look at the console output: You don't really want to have the message repeated two or three times

Reply:
I see, it's mostly about the message, makes sense.